### ISSUE TITLE
[GPU Hardware Texture Compression] Mip maps are not generated

### ISSUE THREAD

**Issue by SilentPhil** [YoYoGames/GameMaker-Bugs#5657]
### Description

I tried out the [new texture compression](https://github.com/YoYoGames/GameMaker-Bugs/issues/5495) feature in my project and found that after enabling BCN compression (specifically BCN -u4, though I haven't tested other configurations), the textures stopped generating Mip Maps.

Compressed textures:
<img width="1282" alt="compressed" src="https://github.com/YoYoGames/GameMaker-Bugs/assets/8656496/cc344755-4339-4115-beed-225f54f62e09">
Not compressed textures:
<img width="1282" alt="not_compressed" src="https://github.com/YoYoGames/GameMaker-Bugs/assets/8656496/0c028b94-bc14-4657-9eb4-bcbfc1aebdde">
RenderDoc - missing mip maps:
<img width="535" alt="missing_mip_maps" src="https://github.com/YoYoGames/GameMaker-Bugs/assets/8656496/b2e2fa76-40b1-4282-b710-daea552f1fe7">

### Steps To Reproduce

Try activating the creation of Mip Maps and texture compression simultaneously.

### How reliably can you recreate this issue using your steps above?

Always

### Which version of GameMaker are you reporting this issue for?

2024.400 (Betas)

### Which platform(s) are you seeing the problem on?

Windows

### Contact Us Package Attached?

[missing_mip_maps.zip](https://github.com/YoYoGames/GameMaker-Bugs/files/15129152/missing_mip_maps.zip)

**Comment by JonnoRobson** [comment]
Typically GameMaker generates mip maps for texture pages at runtime as they are loaded into memory. This process is not feasible for GPU compressed textures as the downsampling must be done on uncompressed data and the process of uncompressing the original data and compressing each mip level is too slow at runtime. 

To solve this I've added an additional step to the asset compilation process that will generate all of the mip levels for texture pages that use GPU compression. There is a tradeoff with this process as storing the additional mip levels will increase the project's on-disk size on a per-texture basis, however there is also the performance benefit of not having to generate the mip maps at runtime.

At the same time I've also fixed a crash when enabling "Separate Texture Page" on textures whose dimensions do not fit the block size of the GPU compression format used. GameMaker will now pad this textures to an appropriate dimension to work with the compression format. Typically this only affects textures created with odd dimensions and only adds a single row or column of pixels so it will not cause unnecessary bloat in texture data.

To be clear these changes will not be in the upcoming August release and will only be available in the next release in October. Once these changes are available we will need to update the appropriate documentation to remove the caveat that textures with "Separate Texture Page" enabled need to be manually padded to be divisible by the block size of the compression format used.
